Assets/LeapMotionModules/UI/Interaction/InteractionManager.cs

```diff
@@ -0,0 +1,293 @@
+    [Tooltip("Whether or not to create the layers used for interaction when the scene runs.")]
+    [SerializeField]
+    protected bool _autoGenerateLayers = true;
+
+    [Tooltip("Layer to use for auto-generation.  The generated interaction layers will have the same collision settings as this layer.")]
+    [SerializeField]
+    protected SingleLayer _templateLayer = 0;
+
+    [Tooltip("The layer containing interaction objects.")]
+    [SerializeField]
+    protected SingleLayer _interactionLayer = 0;
+
+    [Tooltip("The layer containing interaction objects when they become grasped.")]
+    [SerializeField]
+    protected SingleLayer _graspedObjectLayer = 0;
+
+    [Tooltip("The layer containing the colliders for the bones of the hand.")]
+    [SerializeField]
+    protected SingleLayer _contactBoneLayer = 0;
+
+    [SerializeField]
+    #pragma warning disable 0414
+    private bool _showDebugOptions = false;
+    #pragma warning restore 0414
+    [SerializeField]
+    private bool _debugDrawHandGizmos = false;
+
+    #pragma warning disable 0414
+    [HideInInspector]
+    [SerializeField]
+    // Editor-only utility variable for touchActivationRadius property drawing.
+    private bool contactOrGraspingEnabled = true;
+    #pragma warning restore 0414
+
+    /// <summary>
+    /// Provides Frame objects consisting of any and all Hands in the scene. 
+    /// 
+    /// Set by default on Awake(), but can be overridden manually for special situations
+    /// involving multiple LeapServiceProviders.
+    /// </summary>
+    public LeapServiceProvider Provider { get; set; }
+    private float _providerScale = 1F;
+
+    public Action OnGraphicalUpdate = () => { };
+    public Action OnPrePhysicalUpdate = () => { };
+    public Action OnPostPhysicalUpdate = () => { };
+
+    /// <summary>
+    /// Interaction objects further than this distance from a given hand will not be
+    /// considered for any hover interactions with that hand.
+    /// </summary>
+    public float WorldHoverActivationRadius { get { return hoverActivationRadius * _providerScale; } }
+
+    /// <summary>
+    /// Interaction objects further than this distance from a given hand will not be
+    /// considered for any contact or grasping interactions with that hand.
+    /// </summary>
+    public float WorldTouchActivationRadius { get { return touchActivationRadius * _providerScale; } }
+
+    /// <summary>
+    /// A scale that can be used to appropriately transform distances that otherwise expect
+    /// one Unity unit to correspond to one meter.
+    /// </summary>
+    public float SimulationScale { get { return _providerScale; } }
+
+    private InteractionHand[] _interactionHands = new InteractionHand[2];
+    private HashSet<InteractionBehaviourBase> _interactionBehaviours = new HashSet<InteractionBehaviourBase>();
+    private Dictionary<Rigidbody, InteractionBehaviourBase> _rigidbodyRegistry = new Dictionary<Rigidbody, InteractionBehaviourBase>();
```

Comment:
For cases where you want a variable to be publicly gettable but not settable, you can use this syntax

`public Type fieldName { get; private set; }`

This won't work for serialized fields unfortunately because unity cannot serialize properties.